robocop/checkers/misc.py

```diff
@@ -601,6 +607,78 @@
 
         """,
     ),
+    "0927": Rule(
+        RuleParam(
+            name="sections_order",
+            default="documentation,tags,timeout,setup,template,keyword,teardown",
+            converter=parse_test_case_order_param,
+            show_type="str",
+            desc="order of sections in comma-separated list",
+        ),
+        rule_id="0927",
+        name="test-case-section-out-of-order",
+        msg="'{{ section_name }}' is in wrong place of Test Case. "
+        "Recommended order of elements in Test Cases: {{ recommended_order }}",
+        severity=RuleSeverity.WARNING,
+        added_in_version="5.3.0",
+        docs="""
+        Sections should be defined in order set by ``sections_order``
+        parameter (default: ``documentation,tags,timeout,setup,template,keyword,teardown``).
```

Comment:
keyword stands also for any other code like IF, FOR, TRY, comments etc? maybe it should be ``keyword_body``? Or it would be too descriptive

Reply:
Frankly I had problem with theses statements like FOR, IF: they are not type of Token.KEYWORD and they don't have `type` property - this is why I added:
```
            try:
                subnode_type = subnode.type
            except AttributeError:
                continue
```
I was thinking how to handle these statements and I did assumption that in body of TRY / IF / FOR typically there is some keyword, so for example in case of such robot code:
```
[Teardown]    Cleanup
IF    ${something} == ${TRUE}
    Some Keyword
ELSE
    Some Other Keyword
END
```
Lines with with `IF`, `ELSE` will be not reported by the checker but lines inside: `    Some Keyword` and `    Some Other Keyword` will be reported.
Is this enough?

Reply:
Yes, it's good enough. Not always IF/FOR etc blocks contains keywords (it could be statement like RETURN or different block) but this simplification is okay for our purpose